src/IO/H5/VolumeData.cpp

```diff
@@ -0,0 +1,177 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "IO/H5/VolumeData.hpp"
+
+#include <algorithm>
+#include <boost/iterator/transform_iterator.hpp>
+#include <hdf5.h>
+#include <memory>
+#include <ostream>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/TensorData.hpp"
+#include "ErrorHandling/Assert.hpp"
+#include "ErrorHandling/Error.hpp"
+#include "IO/Connectivity.hpp"
+#include "IO/H5/AccessType.hpp"
+#include "IO/H5/Header.hpp"
+#include "IO/H5/Helpers.hpp"
+#include "IO/H5/Version.hpp"
+
+/// \cond HIDDEN_SYMBOLS
+namespace h5 {
+VolumeData::VolumeData(const bool exists, detail::OpenGroup&& group,
+                       const hid_t /*location*/, const std::string& name,
+                       const uint32_t version) noexcept
+    : group_(std::move(group)),
+      name_(name.size() > extension().size()
+                ? (extension() == name.substr(name.size() - extension().size())
+                       ? name
+                       : name + extension())
+                : name + extension()),
+      version_(version),
+      volume_file_root_group_(group_.id(), name_, h5::AccessType::ReadWrite) {
+  if (exists) {
+    const Version open_version(true, detail::OpenGroup{},
+                               volume_file_root_group_.id(), "version");
+    version_ = open_version.get_version();
+    const Header header(true, detail::OpenGroup{}, volume_file_root_group_.id(),
```

Comment:
add error if input version doesn't match version_, (and maybe a comment about backwards compatibility when we have more than one version)